src/main/java/com/mongodb/spark/sql/connector/read/partitioner/AutoBucketPartitioner.java

```diff
@@ -189,6 +191,16 @@ public List<MongoInputPartition> generatePartitions(final ReadConfig readConfig)
       return SINGLE_PARTITIONER.generatePartitions(readConfig);
     }
 
+    Integer serverMaxWireVersion =
+        readConfig.withClient(c -> c.getClusterDescription().getServerDescriptions().stream()
+            .map(ServerDescription::getMaxWireVersion)
+            .max(Integer::compare)
+            .orElse(0));
```

Comment:
If the mapping target is primitive type, usually more specific method is provided. To me the following alternative seems a little bit better:
```
readConfig.withClient(c -> c.getClusterDescription().getServerDescriptions().stream()
            .mapToInt(ServerDescription::getMaxWireVersion)
            .max()
            .orElse(0));
```

but if this is a well-established pattern, it is fine to choose consistency instead of more ideomatic (and efficient) stream usage.

Reply:
That better - thank you - done